resync/api/monitoring_dashboard.py

```
@@ -133,29 +236,43 @@ class DashboardMetricsStore:
-        self._prev_requests = 0
-        self._last_sample_mono = 0.0
-        self._local_lock = asyncio.Lock()
-
-    async def compute_rate_and_add_sample(self, requests_total: int, now_mono: float, sample_builder: Any) -> None:
-        """Calcula RPS localmente e persiste no Redis."""
-        async with self._local_lock:
-            time_delta = now_mono - self._last_sample_mono if self._last_sample_mono > 0 else SAMPLE_INTERVAL_SECONDS
-            req_delta = requests_total - self._prev_requests if self._prev_requests > 0 else 0
+        self._cached_start_time: float | None = None
+
+    async def compute_rate_and_add_sample(self, requests_total: int, now_wall: float, sample_builder: Any) -> None:
```

Ruff (0.15.0)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `sample_builder`

(ANN401)